### ISSUE TITLE
Add regional parameters version to TxRequest

### ISSUE THREAD

**Issue by adriansmares** [TheThingsNetwork/lorawan-stack#4347]
#### Summary

References https://github.com/TheThingsNetwork/lorawan-stack/issues/4196

We should add a `PHYVersion` to `TxRequest`

#### Why do we need this?

As signaled in https://github.com/TheThingsNetwork/lorawan-stack/issues/4196 , data rate indices may map to different SF/BW across different regional parameters version. As such, we need the version in order to allow the Gateway Server to translate the indices to the right values.

#### What is already there? What do you see now?

`rx1_data_rate_index` and `rx2_data_rate_index`, but no `PHYVersion`.

#### What is missing? What do you want to see?

```proto
PHYVersion band_version = 11;
```

The Network Server shall provide the band version, while the Gateway Server shall use them to derive the SF/BW.

#### Environment

`v3.14`

#### How do you propose to implement this?

- Add the proto definitions
- The NS should fill in the version and ID while generating the `TxRequest`
https://github.com/TheThingsNetwork/lorawan-stack/blob/2e5e9dfab39b7adb909f09a7a7e366c57dee6395/pkg/networkserver/downlink.go#L1236
https://github.com/TheThingsNetwork/lorawan-stack/blob/2e5e9dfab39b7adb909f09a7a7e366c57dee6395/pkg/networkserver/downlink.go#L1409
- The PBA should fill these too (https://github.com/packetbroker/api/issues/31)
https://github.com/TheThingsNetwork/lorawan-stack/blob/2e5e9dfab39b7adb909f09a7a7e366c57dee6395/pkg/packetbrokeragent/translation.go#L625
- The GS should take them into account while translating the data rate index
https://github.com/TheThingsNetwork/lorawan-stack/blob/2e5e9dfab39b7adb909f09a7a7e366c57dee6395/pkg/gatewayserver/io/io.go#L440

API changes should target `v3.14.0`, while functional changes may come in subsequent patches.

#### How do you propose to test this?

Test the SF/BW of a downlink sent via AU915 1.0.0 downlink.

#### Can you do this yourself and submit a Pull Request?

I can do the API and NS part, cc @KrishnaIyer for GS and @johanstokking for PBA.

**Comment by johanstokking** [comment]
Cool, @KrishnaIyer please start with API soon, don't close this issue but re-assign once API is there:

- @KrishnaIyer to implement in GS
- @adriansmares to implement in NS
- @johanstokking to implement in PBA (needs https://github.com/packetbroker/api/issues/31)

**Comment by KrishnaIyer** [comment]
> @KrishnaIyer please start with API soon

Isn't API already here: https://github.com/TheThingsNetwork/lorawan-stack/pull/4349?

**Comment by KrishnaIyer** [comment]
This is blocked on https://github.com/TheThingsNetwork/lorawan-stack/issues/4346.

With #4349, we have the `lorawan_phy_version` but the GS can't really use it until https://github.com/TheThingsNetwork/lorawan-stack/issues/4346.

**Comment by johanstokking** [comment]
The band can already be versioned. #4346 is about maintaining the frequency plans in a versioned way, but that won't really change the way GS uses it.

**Comment by KrishnaIyer** [comment]
GS done via #4406. I guess only PBA remains. @johanstokking please close the issue if that's already done.

**Comment by johanstokking** [comment]
PBA still needs to happen
